docs: Added project logo and improved README. (#15)
questdb/rust-maven-plugin: README.md

# rust-maven-plugin

<img src="./artwork/logo_outline_text.svg" alt="rust-maven-plugin">

Build Rust Cargo crates within a Java Maven Project with this plugin.

The use case is to simplify the build process of
[Rust JNI libs](https://crates.io/crates/jni) inside a Java
[Maven](https://maven.apache.org/) project.

# Features

> 

Reviewer: I think it might be worthwhile adding a line introducing the list because the list contains incomplete sentences..

something like:

The plugin provides the following features:

[changing the verb form for each item to calling/informing...etc and so on]

Author: Done.

* Calls `cargo build` as part of a `mvn compile` step.

> * Helpfully informs how to install Rust if can't find it.

Reviewer: ```suggestion
* Helpfully informs how to install Rust if Rust is not found.
```

* Uses Maven's target directory (i.e. `${project.build.directory}`) as the
  `cargo build --target-dir` so `mvn clean` also cleans the Rust crate without
  additional setup in your `pom.xml`.

* Optionally:
  * Can be configured to copy the compiled cdylib(s) to the
    `${project.build.directory}/classes/` directory so the binaries ends up
    bundled in the JAR.

  * As a separate runtime dependency we also provide [`jar-jni`](jar-jni/),
    a helper java library to load the libs from the JAR using a sub-directory
    per `os.name`/`os.arch` so the JAR can contain binaries for multiple
    platforms.

# Status
* Pre-production:

>   * Building, cleaning and bundling into a `.jar` now works.

Reviewer: ```suggestion
  * Building, cleaning, and bundling into a `.jar` now works.
```
